### PR TITLE
ui_test tweaks

### PR DESCRIPTION
**Pull request by RalfJung**
- support multiple filters
- make `./miri check` also cover ui_test
- Run opt-level=4 tests again, but only the "run" tests

r? @oli-obk

### CODE REVIEW

File: ui_test/src/lib.rs [hunk 1]
```diff
@@ -230,6 +235,11 @@ fn run_test(
     }
     let output = miri.output().expect("could not execute miri");
     let mut errors = config.mode.ok(output.status);
+    // Always remove annotation comments from stderr.
+    let annotations = Regex::new(r"\s*//~.*").unwrap();
```
Comment (RalfJung) [inline ui_test/src/lib.rs:239]: Ah, moving this hard-coded normalization here means the test suite does not use it any more... :/

File: ci.sh [hunk 2]
```diff
@@ -26,8 +26,7 @@ function run_tests {
     # optimizations up all the way).
     # Optimizations change diagnostics (mostly backtraces), so we don't check them
     #FIXME(#2155): we want to only run the pass and panic tests here, not the fail tests.
-    #MIRIFLAGS="-O -Zmir-opt-level=4" MIRI_SKIP_UI_CHECKS=1 ./miri test --locked
-    true
+    MIRIFLAGS="-O -Zmir-opt-level=4" MIRI_SKIP_UI_CHECKS=1 ./miri test --locked -- tests/{run-pass,run-fail}
```
[the remark this answers is not in the document]
Comment (RalfJung) [reply ci.sh:29]: Yes.

[the remark this answers is not in the document]
Comment (oli-obk) [reply ci.sh:29]: amazing

**Comment by oli-obk** [comment]
r=me with [80bf204](https://github.com/rust-lang/miri/pull/2166/commits/80bf204848eb2c97effa717e8e4f876f6545d55f) removed

File: ui_test/src/lib.rs (hunk 1 above)
Comment (RalfJung) [reply ui_test/src/lib.rs:262]: Now I am already working on factoring this a bit more so we can test it better. ;)

Comment (RalfJung) [reply ui_test/src/lib.rs:262]: compiletest hard-codes the `//~` normalization as well, right? So why would we remove it?

Comment (oli-obk) [reply ui_test/src/lib.rs:262]: ah ok ^^ I thought just to get the PR to pass quicker

Comment (RalfJung) [reply ui_test/src/lib.rs:262]: So are you find landing this the way it looks now?

Comment (RalfJung) [reply ui_test/src/lib.rs:262]: Ah, I didn't see your other PR, could have resolved this after yours... anyway, thanks!

Comment (oli-obk) [reply ui_test/src/lib.rs:262]: 🤷 was quicker to resolve the conflicts on my end.

**Pull request merged**